resources/assets/lib/beatmap-discussions/editor-toolbar-button.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import * as React from 'react';
+import { isFormatActive, toggleFormat } from './editor-helpers';
+import { SlateContext } from './slate-context';
+
+interface Props {
+  format: string;
+}
+
+export const EditorToolbarButton = (props: Props) => {
+  const context = React.useContext(SlateContext);
+  const handleClick = React.useCallback((event: React.SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    toggleFormat(context, props.format);
+  }, []);
```

Comment:
shouldn't this depend on `props.format`? And maybe `context`? I wonder if it's better to just move it back and read from `dataset` instead.